### PR TITLE
fix(settings): apps list layout

### LINKED ISSUE #42249: [Bug]: Apps UI not working in WebKit browsers (Safari, iOS/iPadOS, GNOME/Epiphany)
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When selection any app in the "your apps" "active apps" "disabled apps" section always the same app will be displayed. E.g. selection "Activity" -> "Versions" is displayed. Scrolling doesn't work as well.
Screenshot 2023-12-13 at 21 53 33

### Steps to reproduce

1. Open Apps
2. Select an app to manipulate
3. in my case "Versions" is displayed

### Expected behavior

The selected app should be displayed to enable/disable

### Installation method

None

### Nextcloud Server version

28

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Nginx

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 22 to 23)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [x] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by ShGKme**
## Summary

Fix Apps Settings layout:
* Store apps layout on mobile: show less columns when sidebar is open
  * Regression from https://github.com/nextcloud/server/pull/44092
* List apps layout on mobile: hide app-level
  * Regression from https://github.com/nextcloud/server/pull/43770
* Refactor apps list settings styles:
  * Remove unused styles used 4..7 years ago
  * Move styles from global `settings.scss` to components
  * Rename classes, follow BEM. Especially with classes like `.section` that were used for both settings sections on other pages and table row
  * Required for the following fix
* Make the search results a valid HTML table
* Resolves: https://github.com/nextcloud/server/issues/42249
  * Regression from https://github.com/nextcloud/server/pull/41439
  * `position: relative` doesn't work on `<tr>` on Safari
  * See: https://bugs.webkit.org/show_bug.cgi?id=240961

### Safari bug

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>

<td>

https://github.com/nextcloud/server/assets/25978914/0c3b078a-6c2b-4995-8bf0-1e86f70164c3

</td>
<td>

https://github.com/nextcloud/server/assets/25978914/9be3cbc2-cd59-451b-8d8e-121a53fcc4a0

</td>

</tr>
</table>

### Other screenshots

Before | After
---|---
Your apps | top
image | image
Your apps | bottom
image | image
Your apps (regression - level is shown) | 600px
image | image
Your apps (regression - level is shown) | 320px (new - actions are hidden)
image | image
App bundles | .
image | image
Store View | 1440px
image | image
Store View (regression - no change with sidebar) | 1440px with sidebar
image | image
Store View | 1280px
image | image
Store View (regression - no change with sidebar) | 1280px with sidebar
image | image
Store View | 600px
image | image
Store View (regression - no change with sidebar) | 600px with sidebar
image | image
Store View | 320px
image | image
Search | no results
image | .
Search (invalid HTML) | Your apps
image | image
image | image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by ShGKme** [comment]
dron is not related

File: apps/settings/src/components/AppList/AppItem.vue [hunk 1]
```diff
@@ -200,21 +208,191 @@ export default {
 </script>
 
 <style scoped lang="scss">
+@use '../../../../../core/css/variables.scss' as variables;
```
Comment (susnux) [inline apps/settings/src/components/AppList/AppItem.vue:211]: They are deprecated it feels not right to still use them as we probably want to get rid of them soon.

**Review by susnux — APPROVED** [review]
Finally the overflow issue is fixed, thank you!

File: apps/settings/src/components/AppList/AppItem.vue (hunk 1 above)
Comment (ShGKme) [reply apps/settings/src/components/AppList/AppItem.vue:211]: Without PostCSS, would you prefer to use just constant numbers in `@media` queries?

**Review by artonge — APPROVED** [review]

**Comment by ShGKme** [comment]
> BTW I think we should in the future separate the table and grid components. That are two different designs and currently the grid one looks a bit "sad".

Yeah. A lot place for further refactoring here. I kept myself from not doing it, to not overload one PR with the fix by refactoring 🙈

**Comment by ShGKme** [comment]
Rebased onto `master`

**Pull request merged**